Ghost_in_the_Shell/client_folder_react_app/ethisim/src/components/EditorComponents/FlowDiagram.js

```diff
@@ -0,0 +1,262 @@
+import React, { useState } from 'react';
+import { makeStyles } from '@material-ui/core/styles';
+import { Container, Typography, Button } from '@material-ui/core';
+import ReactFlow, {
+    Handle,
+    removeElements,
+    isEdge,
+    MiniMap,
+    Controls,
+    Background,
+} from 'react-flow-renderer';
+import { mockComponentsFlowChart } from '../../shared/mockScenarioData';
+
+const useStyles = makeStyles((theme) => ({
+    container: {
+        marginTop: theme.spacing(2),
+        display: 'flex',
+        flexDirection: 'column',
+        alignItems: 'center',
+    },
+    title: {
+        textAlign: 'center',
+    },
+    copyright: {
+        margin: theme.spacing(2),
+    },
+    buttonContainer: {
+        display: 'flex',
+        float: 'right',
+        flexDirection: 'column',
+    },
+    button: {
+        zIndex: 5,
+        float: 'right',
+    },
+}));
+
+const actionNode = ({ data }) => {
+    return (
+        <>
+            <Handle
+                type="target"
+                position="top"
+                onConnect={(params) => console.log('handle onConnect', params)}
+            />
+            {data.label}
+            <Handle
+                type="source"
+                position="bottom"
+                id="a"
+                style={{ left: '20%' }}
+            />
+            <Handle
+                type="source"
+                position="bottom"
+                id="b"
+                style={{ left: '80%' }}
+            />
+        </>
+    );
+};
+
+const reflectionNode = ({ data }) => {
+    return (
+        <>
+            <Handle type="target" position="top" />
+            {data.label}
+            <Handle type="source" position="bottom" />
+        </>
+    );
+};
+
+const genericNode = ({ data }) => {
+    return (
+        <>
+            <Handle type="target" position="top" />
+            {data.label}
+            <Handle type="source" position="bottom" />
+        </>
+    );
+};
+
+const conversationNode = ({ data }) => {
+    return (
+        <>
+            <Handle type="target" position="top" />
+            {data.label}
+            <Handle type="source" position="bottom" />
+        </>
+    );
+};
+
+function initializeElements(componentData) {
+    if (componentData.type === 'Action') {
```

Comment:
Probably a good use case for a switch statement here

Reply:
Done